### PR TITLE
newwindow and clipboard and eventloop::new

### PR DESCRIPTION
**Pull request by gintama91**

### CODE REVIEW

**Comment by gintama91** [comment]
for ruby side see in tests, or in clipboard i commented ruby equivalents for tht file

**Comment by gintama91** [comment]
tests do fail on ci due to cant open display i am not sure how to do them on ci yet(any suggestions for this ?)

File: ext/wry_ruby/src/application/event_loop.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+pub struct SafeWrapper<T> {
+    event_loop: AtomicPtr<T>,
+    _marker: PhantomData<*mut T>,
+}
+
+unsafe impl<T> Send for SafeWrapper<T> {} 
+
+impl EventLoop {
```
Comment (gintama91) [inline ext/wry_ruby/src/application/event_loop.rs:24]: i am not implementing event_loop further(as for now) as per discussion in discord

**Comment by noahgibbs** [comment]
Yeah, headless testing on GTK+ is going to be interesting. People do this kind of thing -- Docker supports a headless mode where it keeps a virtual display, I can see answers about e.g. Sway, a Wayland window manager that will do it.

I'm not sure what we want to use. I haven't done Linux headless GUI testing in a long time, and the tools have changed a lot.

File: ext/wry_ruby/src/application/window.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    application::{
+      event::{Event, StartCause, WindowEvent},
+      event_loop::{ControlFlow},
+    },
+  };
+
+pub fn WindoWnew(title: String, width: u32, height: u32, resizable: bool) {
+    let event_loop = wry::application::event_loop::EventLoop::new();
+    let mut window_builder = WindowBuilder::new();
+    window_builder = window_builder.with_title(title.clone());
```
Comment (gintama91) [inline ext/wry_ruby/src/application/window.rs:15]: hm we prolly dont need to clone here ig

Comment (noahgibbs) [reply ext/wry_ruby/src/application/window.rs:15]: I'd assume the Rust side copies it, yeah.

**Pull request merged**